cinema/permissions.py

```diff
@@ -0,0 +1,9 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    def has_permission(self, request, view):
+        return bool(
+            request.method in SAFE_METHODS and request.user
+            and request.user.is_authenticated
+        ) or (request.user and request.user.is_staff)
```

Comment:
The logic here correctly implements the required permission: allowing safe methods for authenticated users and all actions for admin users. This aligns with the task's requirement to limit actions based on user roles.